parser/ParseBlockProperties.hpp

```diff
@@ -0,0 +1,245 @@
+   * 
+   * @param line_number Line number of the first token of this node in the SQL statement.
+   * @param column_number Column number of the first token of this node in the SQL statement.
+   * @return A caller-managed BlockPropertyItem representing the property of 
+   *         COMPRESS ALL for compressing all attributes in a table.
+   */
+  static ParseBlockPropertyItem* GetCompressAllItem(int line_number, int column_number) {
+    ParseBlockPropertyItem* compress_all = new ParseBlockPropertyItem(line_number,
+                                                                      column_number,
+                                                                      Property::kCompress,
+                                                                      new PtrList<ParseString>());
+    compress_all->compress_all_ = true;
+    return compress_all;
+  }
+
+  /**
+   * @return Name of the parser construct represented by this class.
+   */
+  std::string getName() const override {
+    return std::string("BlockPropertyItem");
+  }
+
+  /**
+   * @brief The type of property being described.
+   * 
+   * @return The specific block property which this instance is describing.
+   */
+  Property property() const {
+    return property_;
+  }
+
+  /**
+   * @return A list of ParseString values which the parser has captured to 
+   *         describe this property. This may be empty in the case of 'COMPRESS ALL'.
+   */
+  const PtrList<ParseString>& values() const {
+    return *(values_.get());
+  }
+
+  /**
+   * @brief Gets a string version of the property which this pair describes.
+   * @details For example, if this the property type of 'kCompress' will return 
+   *          a string object with the value 'compress'.
+   *
+   * @return A string description of the property.
+   */
+  std::string getPropertyString() const;
+
+  /**
+   * @brief Returns if the user specified to compress all attributes (COMPRESS 
+   *  ALL) attributes using a special keyword.
+   *  
+   * @return \ctrue if this is a compression property and the value is 'ALL'.
+   */
+  bool compressAll() const;
+
+ protected:
+  void getFieldStringItems(
+      std::vector<std::string> *inline_field_names,
+      std::vector<std::string> *inline_field_values,
+      std::vector<std::string> *non_container_child_field_names,
+      std::vector<const ParseTreeNode*> *non_container_child_fields,
+      std::vector<std::string> *container_child_field_names,
+      std::vector<std::vector<const ParseTreeNode*>> *container_child_fields) const override {
+    inline_field_names->push_back("property");
+    inline_field_values->push_back(getPropertyString());
+
+    if (values_->size() > 1) {
+      container_child_field_names->push_back(std::string("values"));
+      container_child_fields->emplace_back();
+      for (const ParseString& item_value : *values_) {
+        container_child_fields->back().push_back(&item_value);
+      }
+    } else if (values_->size() == 1) {
```

Comment:
I think a single value should also be treated as a (single element) list using `container_child_fields` instead of this special case.